pkg/handler/frontend-config.go

```diff
@@ -0,0 +1,36 @@
+package handler
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"gopkg.in/yaml.v2"
+)
+
+type frontendConfig struct {
+	PortNaming struct {
+		Enable    bool              `yaml:"Enable,omitempty" json:"Enable,omitempty"`
+		PortNames map[string]string `yaml:"PortNames,omitempty" json:"PortNames,omitempty"`
+	} `yaml:"PortNaming,omitempty" json:"PortNaming,omitempty"`
+}
+
+func GetConfig(FrontendConfig string) func(w http.ResponseWriter, r *http.Request) {
+	cfg := frontendConfig{}
+	yamlFile, err := ioutil.ReadFile(FrontendConfig)
+	resp := []byte{}
+	if err == nil {
+		err = yaml.Unmarshal(yamlFile, &cfg)
+		if err == nil {
+			resp, err = json.Marshal(cfg)
+		}
+	}
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		if err != nil {
+			writeError(w, http.StatusInternalServerError, err.Error())
```

Comment:
I there was any temporary error... should we retry the read on each invocation instead of returning always error?

Reply:
Ok, since I would like to keep caching part I will update to the following behavior: this will retry as long as it get an error but will cache the first success.